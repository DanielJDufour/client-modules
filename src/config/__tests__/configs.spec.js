const configs = require('../index');

describe('Configs: ', () => {

  it('match snapshots', () => {
    const allConfigs = Object.keys(configs).map((config) => {
      return configs[config]({
        context: __dirname
      });
    });
    expect(allConfigs).toMatchSnapshot();
  })

});
