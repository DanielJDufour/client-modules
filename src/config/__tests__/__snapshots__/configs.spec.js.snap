exports[`Configs:  match snapshots 1`] = `
Array [
  Object {
    "entry": "/Users/jake/Sites/Codecademy/npm/webpack-config/src/config/__tests__/src/main.js",
    "module": Object {
      "rules": Array [
        Object {
          "loader": "babel-loader",
          "options": Object {
            "cacheDirectory": true,
          },
          "test": /\\.js?$/,
        },
        Object {
          "loader": "file-loader",
          "test": /\\.jpe?g$|\\.gif$|\\.png$|\\.svg$|\\.woff$|\\.ttf$|\\.eot|\\.wav$|\\.mp3$/,
        },
      ],
    },
    "output": Object {
      "filename": "[name].js",
      "path": "/Users/jake/Sites/Codecademy/npm/webpack-config/src/config/__tests__/dist",
    },
    "plugins": Array [
      DefinePlugin {
        "definitions": Object {
          "__DEV__": true,
          "process.env.NODE_ENV": "\"development\"",
        },
      },
      ContextReplacementPlugin {
        "newContentRecursive": undefined,
        "newContentRegExp": /en/,
        "newContentResource": undefined,
        "resourceRegExp": /moment[\\/\\\\]locale$/,
      },
    ],
    "resolve": Object {
      "extensions": Array [
        ".webpack.js",
        ".web.js",
        ".js",
        ".scss",
        ".json",
      ],
      "modules": Array [
        "node_modules",
        "node_modules/@codecademy",
      ],
    },
  },
  Object {
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\.css?$/,
          "use": Array [
            Object {
              "loader": "css-loader/locals",
              "options": Object {
                "importLoaders": 1,
                "localIdentName": "[name]__[local]___[hash:base64:5]",
                "sourceMap": true,
              },
            },
          ],
        },
        Object {
          "test": /\\.scss?$/,
          "use": Array [
            Object {
              "loader": "css-loader/locals",
              "options": Object {
                "importLoaders": 1,
                "localIdentName": "[name]__[local]___[hash:base64:5]",
                "modules": true,
                "sourceMap": true,
              },
            },
            Object {
              "loader": "postcss-loader",
              "options": Object {
                "plugins": [Function],
              },
            },
            Object {
              "loader": "sass-loader",
              "options": Object {
                "sourceMap": true,
              },
            },
          ],
        },
      ],
    },
    "output": Object {
      "libraryTarget": "umd",
    },
  },
  Object {
    "devtool": "#cheap-source-map",
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\.css?$/,
          "use": Array [
            Object {
              "loader": "style-loader",
            },
            Object {
              "loader": "css-loader",
              "options": Object {
                "importLoaders": 1,
                "localIdentName": "[name]__[local]___[hash:base64:5]",
                "sourceMap": true,
              },
            },
            Object {
              "loader": "postcss-loader",
              "options": Object {
                "plugins": [Function],
              },
            },
          ],
        },
        Object {
          "test": /\\.scss?$/,
          "use": Array [
            Object {
              "loader": "style-loader",
            },
            Object {
              "loader": "css-loader",
              "options": Object {
                "importLoaders": 1,
                "localIdentName": "[name]__[local]___[hash:base64:5]",
                "modules": true,
                "sourceMap": true,
              },
            },
            Object {
              "loader": "postcss-loader",
              "options": Object {
                "plugins": [Function],
              },
            },
            Object {
              "loader": "sass-loader",
              "options": Object {
                "sourceMap": true,
              },
            },
          ],
        },
      ],
    },
    "output": Object {
      "devtoolFallbackModuleFilenameTemplate": "[resourcePath]?[hash]",
      "devtoolModuleFilenameTemplate": "[absolute-resource-path]",
    },
  },
  Object {
    "devServer": Object {
      "headers": Object {
        "Access-Control-Allow-Origin": "*",
      },
      "port": 3808,
      "publicPath": "http://localhost:3808/dist/",
      "stats": Object {
        "assets": false,
        "chunkModules": false,
        "chunks": false,
        "colors": true,
        "hash": false,
        "timings": true,
        "version": false,
      },
    },
    "module": Object {
      "rules": Array [
        Object {
          "loader": "babel-loader",
          "options": Object {
            "plugins": Array [
              "react-transform",
              Object {
                "transforms": Array [
                  Object {
                    "imports": Array [
                      "react",
                    ],
                    "locals": Array [
                      "module",
                    ],
                    "transform": "react-transform-hmr",
                  },
                ],
              },
            ],
          },
        },
      ],
    },
    "plugins": Array [
      NoEmitOnErrorsPlugin {},
    ],
  },
  Object {
    "devtool": false,
    "plugins": Array [
      LoaderOptionsPlugin {
        "options": Object {
          "minimize": true,
          "test": Object {
            "test": [Function],
          },
        },
      },
      UglifyJsPlugin {
        "options": Object {},
      },
    ],
  },
  Object {
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\.css?$/,
          "use": Array [
            Object {
              "loader": "/Users/jake/Sites/Codecademy/npm/webpack-config/node_modules/extract-text-webpack-plugin/loader.js",
              "options": Object {
                "omit": 1,
                "remove": true,
              },
            },
            Object {
              "loader": "style-loader",
            },
            Object {
              "loader": "css-loader",
              "options": Object {
                "importLoaders": 1,
                "localIdentName": "[name]__[local]___[hash:base64:5]",
                "sourceMap": true,
              },
            },
            Object {
              "loader": "postcss-loader",
              "options": Object {
                "plugins": [Function],
              },
            },
          ],
        },
        Object {
          "test": /\\.scss?$/,
          "use": Array [
            Object {
              "loader": "/Users/jake/Sites/Codecademy/npm/webpack-config/node_modules/extract-text-webpack-plugin/loader.js",
              "options": Object {
                "omit": 1,
                "remove": true,
              },
            },
            Object {
              "loader": "style-loader",
            },
            Object {
              "loader": "css-loader",
              "options": Object {
                "importLoaders": 1,
                "localIdentName": "[name]__[local]___[hash:base64:5]",
                "modules": true,
                "sourceMap": true,
              },
            },
            Object {
              "loader": "postcss-loader",
              "options": Object {
                "plugins": [Function],
              },
            },
            Object {
              "loader": "sass-loader",
              "options": Object {
                "sourceMap": true,
              },
            },
          ],
        },
      ],
    },
    "plugins": Array [
      ExtractTextPlugin {
        "filename": "[name].css",
        "id": 1,
        "options": Object {
          "allChunks": false,
        },
      },
    ],
  },
]
`;
